Dapper/SqlMapper.cs

```diff
@@ -2000,7 +2005,7 @@ private static Exception MultiMapException(IDataRecord reader, string splitOnCol
                         values[0] = r.GetValue(startBound);
                         if (values[0] is DBNull)
                         {
-                            return null;
+                            return null!;

```

Comment:
IMO this isn't accurate...but it's also viral, realistically we have a `Func<DbDataReader, object?>` throughout here - thoughts?

Reply:
It is a tricky one, because then it makes the core APIs things like `IEnumerable<T?> Query<T>` - which is wrong more than it is right. The exact nuance isn't expressable, sadly